### PR TITLE
Add flags for metadata options

### PR DESCRIPTION
**Pull request by aiyengar2**
Introduces a `ModifyInstanceMetadataOptions` call after the instances are running to inject the user provided `HttpEndpoint` and `HttpTokens` metadata flags if requested by the user.

AWS SDK update is required as `ModifyInstanceMetadataOptions` was not available in the older version rancher/machine was previously using.

https://docs.aws.amazon.com/cli/latest/reference/ec2/modify-instance-metadata-options.html

### CODE REVIEW

**Comment by aiyengar2** [comment]
Code is not ready to be merged; currently receiving the following error on trying to provision either a spot instance or a regular instance:

```
Error creating machine: Error detecting OS: Too many retries waiting for SSH to be available.  Last error: Maximum number of retries (60) exceeded, requeuing
```

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/25078

**Comment by aiyengar2** [comment]
> Code is not ready to be merged; currently receiving the following error on trying to provision either a spot instance or a regular instance:
> 
> ```
> Error creating machine: Error detecting OS: Too many retries waiting for SSH to be available.  Last error: Maximum number of retries (60) exceeded, requeuing
> ```
> 
> When the flags are removed, the error disappears and a node is able to be provisioned correctly as expected.
> 
> This may have something to do with the `httpTokens: required` option as the default nodes also have `httpEndpoint: enabled`.

This appears to be an issue with the default AMI I used, which likely may not have the required updated dependencies to support using EC2 IMDSv2. Switched the default Ubuntu AMI on Rancher to the latest release in us-west-2 `ami-05d8a2e90265271e5` and the error disappeared.

Code should be ready for review / merge now.

Note: EC2 IMDSv2 is uses session based authentication, which is exactly what the flags above seek to enforce using with `httpTokens: required`.

File: drivers/amazonec2/amazonec2.go [hunk 1]
```diff
@@ -752,6 +784,17 @@ func (d *Driver) innerCreate() error {
 
 	d.waitForInstance()
 
+	if d.HttpEndpoint != "" && d.HttpTokens != "" {
```
Comment (aiyengar2) [inline drivers/amazonec2/amazonec2.go:787]:
```suggestion
	if d.HttpEndpoint != "" || d.HttpTokens != "" {
```

Comment (aiyengar2) [reply drivers/amazonec2/amazonec2.go]: updated in latest commit

**Pull request merged**